exercises/acronym/example.cpp

```diff
@@ -0,0 +1,26 @@
+#include <string>
+#include <vector>
+#include <boost/algorithm/string.hpp>
+#include "acronym.h"
+
+using std::string;
+using std::vector;
+
+namespace acronym {
+
+const string acronym(string const& input_str)
+{
+    vector<string> words;
+    boost::split(words, input_str, boost::is_any_of(" :-,"));
+
+    string acronym_str;
+    for(vector<string>::iterator it = words.begin(); it != words.end(); ++it)
```

Comment:
You probably want to use a ranged for loop here. [See here, especially the examples.](http://en.cppreference.com/w/cpp/language/range-for)

I think you want:
```
for (auto const& s: words)
    acronym_str.push_back(to_upper(s.front()));
```

Reply:
Yeah, my version is fairly verbose, admittedly. I will change this to a range-for instead, as you suggested.